### PR TITLE
feat: support branch deploy

### PR DESCRIPTION
**Pull request by nwtgck**

### CODE REVIEW

**Comment by rajington** [comment]
i think this would need to be upgraded https://github.com/nwtgck/actions-netlify/blob/feature/branch-deploy/package.json#L29

File: src/main.ts [hunk 1]
```diff
@@ -82,7 +87,8 @@ async function run(inputs: Inputs): Promise<void> {
     const deploy = await netlifyClient.deploy(siteId, deployFolder, {
       draft: isDraft,
       message: deployMessage,
-      configPath: netlifyConfigPath
+      configPath: netlifyConfigPath,
+      branch: getBranchName() // https://github.com/netlify/js-client/releases/tag/v4.3.0
```
Comment (rajington) [inline src/main.ts:91]: could this also take in the branch from inputs?
`const branch: string | undefined = inputs.branch()`
```suggestion
      branch: branch || getBranchName() // https://github.com/netlify/js-client/releases/tag/v4.3.0
```

**Comment by nwtgck** [comment]
@rajington Thanks. I didn't merge the pull request below.
https://github.com/nwtgck/actions-netlify/pull/175

Yes, I'll make an input for branch deploy feature. This PR is a kind of WIP.

**Comment by rajington** [comment]
submitted #178 with updated deps/docs/tests

**Comment by nwtgck** [comment]
Superseded by #178

**Pull request closed without merging**